Rubberduck.VBEEditor/Events/VbeEvents.cs

```diff
@@ -0,0 +1,227 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Rubberduck.VBEditor.SafeComWrappers;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+
+namespace Rubberduck.VBEditor.Events
+{
+    public sealed class VBEEvents : IVBEEvents
+    {
+        private static VBEEvents _instance;
```

Comment:
Why a static singleton instead of constructor injection and registration in the IoC container in singleton scope?

Reply:
Because the ctor is not parameterless and a normal singleton can only have parameterless static constructor. C# does not permit a static constructor to contain parameters. Furthermore, we need to be able to deterministically shut down all event notifications, which has to be done via a static method.

Reply:
First, you can register as singleton in the IoC container with any constructor you want.

The second point is an actual argument not to let the IoC container deal with the disposal, but I do not think we need a static method for it. Instead, we could construct the event provider immediately before the IoC installer and let the IoC container inject that instance, just like we do with the VBE and AddIn instances. Then, we have access to it in the shutdown code and can terminate the instance there.

Well, I guess it also works with this static approach, but I have to say that I am not a fan of static code.

Reply:
Thanks for mentioning the `RegisterConstantVbeAndAddIn` -- I can see more clearly how I could have done this using IoC which wasn't clear to me before. 

Reviewing the code again, I see that the only other place the `Terminate` would have been called was in the source code presenter which we already removed from the earlier PR (and even so it did not feel right to me). I will revise to allow use of IoC in similar manner to the VBE/AddIn so it becomes consistent. 

I only have one more concern. if I remove the private ctor and allow it to be constructed by others, what is there to prevent misuse of the class since multiple instance would destroy its ability to deterministically clean up events?